Update contributing guidelines (#155)
nordcraftengine/documentation: docs/18-guides/99-contribution/index.md

## Documentation structure

The Nordcraft documentation is organized into sections and individual pages:

- **Sections**: Main categories like "Get started," "Components," or "Styling"
- **Pages**: Individual documents that can be within sections or standalone (like the "About" page)

Each section and page corresponds to a folder in the [GitHub repository](https://github.com/nordcraftengine/documentation):

- Sections and pages are numbered (01, 02, 03...) to determine their order in the side menu
- The content of each page is stored in an `index.md` markdown file
- All related images are placed in the same folder as the page's `index.md` file

## Creating a new page

### File setup

1. Identify which section your page belongs to
2. Create a new folder with the appropriate number (to maintain order) and name
   - For example: `05-the-editor` for the main section
   - Or `05-the-editor/03-bottom-bar` for a page within a section
3. Create an `index.md` file within this folder

## Required structure

Each documentation page needs:

1. **YAML frontmatter** at the top of the file with:

>    - `title`: Page title in sentence case (used for the SEO title)

Reviewer: ```suggestion
   - `title`: Page title in sentence case (used for the page's title)
```

>    - `description`: Brief overview of the page's content (used for the SEO metadescription, keep this to under 155 characters if possible)

Reviewer: ```suggestion
   - `description`: Brief overview of the page's content (used for the page's meta description, keep this to under 155 characters if possible)
```

### Page structure

1. **Introduction**: Begin with a brief explanation of the feature or concept
2. **Main content**: Explain key concepts with clear examples or images
3. **Related information**: Link to relevant documentation throughout the page where appropriate

Each page should be self-contained and understandable on its own, while still connecting to the broader documentation through relevant links.

### Common sections

Depending on the page type, consider including:

- **Definition**: What is this feature?
- **Purpose**: Why would someone use it?
- **Example**: How does it work in practice?
- **Steps**: How to implement it
- **Best practices**: Tips for optimal usage
- **Common issues**: Troubleshooting guidance

### Writing style

- **Clear and concise**: Use simple, direct language
- **Formal but approachable**: Professional but not academic
- **Action-oriented**: Focus on how to use features
- **User-focused**: Address the reader directly ("you")

Keep in mind the following when updating or adding documentation:

- Do not use unnecessary words or marketing language
- Do not include placeholder text or TODOs
- Do not reference personal opinions or subjective assessments

> - Do not create circular references or link to non-existent pages

Reviewer: ```suggestion
- Do not create circular references or links to non-existent pages
```
